lib/chef_handler_foreman/foreman_facts.rb

```diff
@@ -85,11 +108,26 @@ def get_key(key, prefix)
     end
 
     def send_attributes(attributes)
-      if uploader
-        uploader.foreman_request('/api/hosts/facts', attributes, node.name)
+      if @cachefile and !@@cache_expired
+        Chef::Log.info "No attributes have changed - not uploading to foreman" 
       else
-        Chef::Log.error "No uploader registered for foreman facts, skipping facts upload"
+        if uploader
+          uploader.foreman_request('/api/hosts/facts', attributes, node.name)
+        else
+          Chef::Log.error "No uploader registered for foreman facts, skipping facts upload"
+        end
+      end
+    end
+
+    def verify_checksum(attributes)
+      attrs_checksum = Digest::MD5.hexdigest(attributes.to_s)
+      if File.exist?(@cachefile)
+        contents = File.open(@cachefile, 'r') { |f| f.read }
+        if attrs_checksum.eql?(contents)
+          @@cache_expired = false
```

Comment:
I think there's no need to use class variable here (double @) as it's accessed only within the instance (and I prefer not using class variables if possible). Could you replace `@@cache_expired` with `@cache_expired` throughout the file? Also adding attr_accessor :cached_expired would be nice.